From 08e54692dfbc89288971c416f4da7023db406133 Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Mon, 23 Jun 2014 19:58:15 +0200
Subject: [PATCH 242/290] gpu: drm: xylon: Removed unused code from logicvc
 driver

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Removed unused or unneeded code from logicvc driver files.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
(cherry picked from commit 4b927aa4620636bfe7ad822462bae7a6ea57312f)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xylon/xylon_logicvc.c       | 24 ------------------------
 drivers/gpu/drm/xylon/xylon_logicvc_hw.h    |  1 -
 drivers/gpu/drm/xylon/xylon_logicvc_layer.h |  1 -
 3 files changed, 26 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_logicvc.c b/drivers/gpu/drm/xylon/xylon_logicvc.c
index 040d08d..f05e83f 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc.c
+++ b/drivers/gpu/drm/xylon/xylon_logicvc.c
@@ -145,14 +145,6 @@
 #define LOGICVC_DTYPE_REG_INIT 0
 
 /*
- * logiCVC display power signals
- */
-#define LOGICVC_EN_BLIGHT_MSK (1 << 0)
-#define LOGICVC_EN_VDD_MSK    (1 << 1)
-#define LOGICVC_EN_VEE_MSK    (1 << 2)
-#define LOGICVC_V_EN_MSK      (1 << 3)
-
-/*
  * logiCVC various definitions
  */
 #define LOGICVC_MAJOR_REVISION_SHIFT 11
@@ -162,7 +154,6 @@
 #define LOGICVC_PATCH_LEVEL_MASK     0x1F
 
 #define LOGICVC_LAYER_ON        (1 << 0)
-#define LOGICVC_SWAP_RB         (1 << 4)
 #define LOGICVC_PIX_DATA_INVERT (1 << 7)
 #define LOGICVC_PIX_ACT_HIGH    (1 << 8)
 #define LOGICVC_MIN_HRES         64
@@ -171,7 +162,6 @@
 #define LOGICVC_MAX_VRES         2048
 #define LOGICVC_MAX_LINES        4096
 #define LOGICVC_MAX_LAYERS       5
-#define LOGICVC_CLUT_SIZE        256
 
 #define LOGICVC_READABLE_REGS    (1 << 0)
 #define LOGICVC_SIZE_POSITION    (1 << 1)
@@ -333,13 +323,6 @@ unsigned int xylon_cvc_get_layers_num(struct xylon_cvc *cvc)
 	return cvc->layers;
 }
 
-unsigned int xylon_cvc_get_layers_max_width(struct xylon_cvc *cvc)
-{
-	struct xylon_cvc_layer_data *layer_data = cvc->layer_data[0];
-
-	return layer_data->fix_data.width;
-}
-
 u32 xylon_cvc_layer_get_format(struct xylon_cvc *cvc, int id)
 {
 	struct xylon_cvc_layer_data *layer_data = cvc->layer_data[id];
@@ -656,13 +639,6 @@ void xylon_cvc_int_free(struct xylon_cvc *cvc, void *dev)
 	free_irq(cvc->irq, dev);
 }
 
-void xylon_cvc_reset(struct xylon_cvc *cvc)
-{
-	void __iomem *base = cvc->base;
-
-	writel(LOGICVC_DTYPE_REG_INIT, base + LOGICVC_DTYPE_ROFF);
-}
-
 void xylon_cvc_enable(struct xylon_cvc *cvc, struct videomode *vmode)
 {
 	void __iomem *base = cvc->base;
diff --git a/drivers/gpu/drm/xylon/xylon_logicvc_hw.h b/drivers/gpu/drm/xylon/xylon_logicvc_hw.h
index 97254a2..19ac46e 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc_hw.h
+++ b/drivers/gpu/drm/xylon/xylon_logicvc_hw.h
@@ -41,7 +41,6 @@ int xylon_cvc_int_request(struct xylon_cvc *cvc, unsigned long flags,
 			  irq_handler_t handler, void *dev);
 void xylon_cvc_int_free(struct xylon_cvc *cvc, void *dev);
 
-void xylon_cvc_reset(struct xylon_cvc *cvc);
 void xylon_cvc_enable(struct xylon_cvc *cvc, struct videomode *vmode);
 void xylon_cvc_disable(struct xylon_cvc *cvc);
 
diff --git a/drivers/gpu/drm/xylon/xylon_logicvc_layer.h b/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
index de1e704..b3d5c02 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
+++ b/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
@@ -28,7 +28,6 @@ enum xylon_cvc_layer_ctrl_cmd {
 struct xylon_cvc;
 
 unsigned int xylon_cvc_get_layers_num(struct xylon_cvc *cvc);
-unsigned int xylon_cvc_get_layers_max_width(struct xylon_cvc *cvc);
 
 u32 xylon_cvc_layer_get_format(struct xylon_cvc *cvc, int id);
 unsigned int xylon_cvc_layer_get_bits_per_pixel(struct xylon_cvc *cvc, int id);
-- 
2.0.2

